pkg/operator/controller/ingress/status.go

```diff
@@ -171,45 +184,113 @@ func computeDeploymentPodsScheduledCondition(deployment *appsv1.Deployment, pods
 }
 
 // computeIngressAvailableCondition computes the ingress controller's current Available status state
-// by inspecting the Available condition of deployment. The ingresscontroller is only available if
-// the deployment is also available.
-func computeIngressAvailableCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
-	for _, cond := range deployment.Status.Conditions {
-		if cond.Type != appsv1.DeploymentAvailable {
+// by inspecting the following:
+// 1) the Available condition of Deployment,
+// 2) the DNSReady condition of the IngressController, and
+// 3) the LoadBalancerReady condition of the IngressController.
+// The ingresscontroller is judged Available only if all 3 conditions are true
+func computeIngressAvailableCondition(conditions []operatorv1.OperatorCondition) operatorv1.OperatorCondition {
+
+	expected := []expectedCondition{
+		{
+			condition: IngressControllerDeploymentAvailableConditionType,
+			status:    operatorv1.ConditionTrue,
+			mustHave:  true,
+		},
+		{
+			condition: operatorv1.DNSReadyIngressConditionType,
+			status:    operatorv1.ConditionTrue,
+			ifConditionsTrue: []string{
+				operatorv1.LoadBalancerManagedIngressConditionType,
+				operatorv1.LoadBalancerReadyIngressConditionType,
+				operatorv1.DNSManagedIngressConditionType,
+			},
+			gracePeriod: time.Second * 30,
+			mustHave:    true,
+		},
+		{
+			condition:        operatorv1.LoadBalancerReadyIngressConditionType,
+			status:           operatorv1.ConditionTrue,
+			ifConditionsTrue: []string{operatorv1.LoadBalancerManagedIngressConditionType},
+			gracePeriod:      time.Second * 90,
+			mustHave:         true,
```

Comment:
Specifying `mustHave: true` for `operatorv1.DNSReadyIngressConditionType` and `operatorv1.LoadBalancerReadyIngressConditionType` is not going to work properly when the endpoint publishing strategy type is not `LoadBalancerService` (or if DNS management is turned off by clearing the zones in the cluster DNS config object) because `computeLoadBalancerStatus` and `computeDNSStatus` do not set the `LoadBalancerReady` or `DNSReady` status conditions if the operator is not managing an LB or DNS.

Reply:
You're right, I noticed it had an adverse affect on the end to end testing.  I have removed this construct and replaced with a check for empty conditions.